servers/rendering/renderer_rd/renderer_scene_render_rd.h

```diff
@@ -77,13 +80,161 @@ struct RenderDataRD {
 	bool directional_light_soft_shadows = false;
 
 	RenderingMethod::RenderInfo *render_info = nullptr;
+
+	/* Shadow data */
+	const RendererSceneRender::RenderShadowData *render_shadows = nullptr;
+	int render_shadow_count = 0;
+
+	LocalVector<int> cube_shadows;
+	LocalVector<int> shadows;
+	LocalVector<int> directional_shadows;
+
+	/* GI info */
+	const RendererSceneRender::RenderSDFGIData *render_sdfgi_regions = nullptr;
+	int render_sdfgi_region_count = 0;
+	const RendererSceneRender::RenderSDFGIUpdateData *sdfgi_update_data = nullptr;
+
+	uint32_t voxel_gi_count = 0;
+};
+
+struct SceneLightDataRD {
+	enum {
```

Comment:
IMO all this should be moved to LightStorage (decal to TextureStorage?) there is not much of a reason to unify thesesince its just an array that you need to pass to the render shader, which can simply be obtained with a function and which will not change from frame to frame.

Reply:
Agreed, I shall work on that. As I mentioned in chat, should we do this as a separate PR and merge the work already done? This PR is getting fairly large.

Reply:
This is now completed, `SceneLightDataRD` is no more! All data structures and code are now in `LightStorage` and `TextureStorage`.

These buffers are depth sorted and distance fade is applied so I do think it currently updates frame by frame but all in all, it is all a lot cleaner now.

There are still a few ties to the render implementation that could be improved but I think those are not critical.